### DISCUSSION TITLE
my Roborock S5 goes to another room

### DISCUSSION THREAD

**Discussion by maciek188** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#500] [Q&A] [answered]
when I choose Bedroom my roborock goes  to the orange room instead (and clean it perfectly all [here I manually stoped it] ).

How this is even possible? where is the mistake?

when I do "copy service call" I got:

```json
{
  "domain": "xiaomi_miio",
  "service": "vacuum_clean_segment",
  "serviceData": {
    "segments": [
      1
    ],
    "entity_id": "vacuum.roborock"
  }
}
```

![Screenshot 2022-11-18 165800](https://user-images.githubusercontent.com/96416174/202747483-e5520869-4619-4c27-b2f0-75a91a19a57d.png)

**Comment by PiotrMachowski** [comment, marked as answer]
Have you set up room cleaning according to [this instruction](https://github.com/PiotrMachowski/lovelace-xiaomi-vacuum-map-card/discussions/317)?

**Reply by maciek188** [reply]
Thank you for your reply,
I did it "manually" before the new version of xiaomi-vacuum-map-card comes.
At the end what I need is to clean a specific room triggered by HA automations.

I will try to define the rooms according to the latest instruction.

Thank you

**Reply by PiotrMachowski** [reply]
Ok, so are you sure that room numbers are correct?

**Comment by maciek188** [comment]
I started with 3 rooms only. The orange room was not even defined.

With the newest intructions now it works, thank you again for your help! 🥇   

![Screenshot 2022-11-18 185809](https://user-images.githubusercontent.com/96416174/202771535-44afde7c-9a73-4541-a979-0899c549439b.png)

btw. I have a nice idea for the new function: 

when mop attached avoid carpets 

(but I don't  know how to do a request on github - pull request ask me to campare something)

Have a great evening!

**Reply by PiotrMachowski** [reply]
> btw. I have a nice idea for the new function:
> 
> when mop attached avoid carpets

It seems more like an automation than something that should be done in the card.
